storage/innobase/fil/fil0crypt.cc

```diff
@@ -411,22 +421,16 @@ static ulint fsp_header_get_keyring_encryption_offset(
   return offset;
 }
 
-fil_space_crypt_t *fil_space_read_crypt_data(const page_size_t &page_size,
-                                             const byte *page) {
-  ulint bytes_read = 0;
-
-  const ulint offset = fsp_header_get_keyring_encryption_offset(page_size);
+static fil_space_crypt_t *fil_space_read_crypt_data_v1(
+    const page_size_t &page_size, const byte *page) {
+  const ulint offset{fsp_header_get_keyring_encryption_offset(page_size)};
 
-  if (memcmp(page + offset, ENCRYPTION_KEY_MAGIC_PS_V1,
-             ENCRYPTION_MAGIC_SIZE) != 0) {
-    /* Crypt data is not stored. */
-    return NULL;
-  }
+  ut_ad(memcmp(page + offset, ENCRYPTION_KEY_MAGIC_PS_V1,
+               ENCRYPTION_MAGIC_SIZE) == 0);
 
-  bytes_read += ENCRYPTION_MAGIC_SIZE;
+  ulint bytes_read{ENCRYPTION_MAGIC_SIZE};
 
-  uint8_t iv_length = mach_read_from_2(page + offset + bytes_read);
-  bytes_read += 2;
+  bytes_read += 2;  // skip iv_length
```

Comment:
For v1, why do we now skip iv_length check ?

Reply:
it is always 16 bytes, so no need to read this ... do you want me to ?

Reply:
better to read and assert that it is 16 bytes. It can help in catching corruption

Reply:
done